crates/iceberg/src/error.rs

```diff
@@ -44,6 +44,8 @@ pub enum ErrorKind {
     ///
     /// This error is returned when given iceberg feature is not supported.
     FeatureUnsupported,
+    /// This error is returned when we try to convert a JSON null value to an iceberg value
```

Comment:
I believe this error kind also need to be merged into `DataInvalid`. The action user can take is the same.

Reply:
I currently need a separate `ErrorKind` because I need to handle the Error in `to_optional_literal`. If we go with a solution that doesn't require `to_optional_literal`, I will remove the ErrorKind again.